### DISCUSSION TITLE
Process crashed Simulation process failed. Exited with code 0xffffffffc0000005.

### DISCUSSION THREAD

**Discussion by Arinomo** [OpenModelica/OpenModelica#9666] [Q&A]
I'm trying to run a simulation with `Start Time = 10` and `Stop Time = 20` and use a previous simulation result and set the `Initialization Mode = none` and `Equation System Initialization Time = 10`. The process is always crashed. 

<details><summary> Some info </summary>

`D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim/ContinuedSim_second_run_w_init_file.exe -port=52232 -logFormat=xmltcp -override=startTime=10,stopTime=20,stepSize=0.001,tolerance=1e-6,solver=dassl,outputFormat=mat,variableFilter=.* -r=D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim/foo.mat -iim=none -iif=ContinuedSim_first_run_res.mat -iit=10 -w -lv=LOG_STATS -inputPath=D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim -outputPath=D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim`
</details>

What does the error code mean?

on OpenModelica v1.19.0-dev-590-g0897872ac1 on Win10

Edit:
The simulation was initialized and running until ca. 19% progress and then crashed with above error code

**Comment by adrpo** [comment]
Did you set the PATH? Looks like DLL not found issue.

`set PATH=%PATH%;%OPENMODELICAHOME%\bin`

Then run your executable.

**Reply by Arinomo** [reply]
I added the OM bin path to windows env variables but the same crash happened. I set the options and run the simulation directly from OMEdit.

**Reply by Arinomo** [reply]
One thing i noticed is, after selecting the `Initialization File` from `Browse` i had to omit the absolute path to file from `D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim/ContinuedSim_first_run_res.mat` to `ContinuedSim_first_run_res.mat`, otherwise OM is unable to read the input file `unable to read input-file <D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim/D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim/ContinuedSim_first_run_res.mat> [Invalid argument]`

**Reply by adeas31** [reply]
@mahge can you please fix this. Seems like simulation runtime is always expecting a relative path.

**Reply by mahge** [reply]
This might have been fixed already. I will check.

**Reply by mahge** [reply]
Okay I don't think this has been fixed. However, the expected behavior should be clarified a bit as well. 
This, I assume, is what we want:

- If an input file (`-iif`) is specified, we look for it directly. Using whatever name/path is given to the option. 
- If cannot be found, we then look for it `relative` to the input path (`-inputPath`)
- If it cannot be found it is an error.

I think this is doable without a lot of complications.

**Reply by mahge** [reply]
My bad. @AnHeuermann has already fixed this in https://github.com/OpenModelica/OpenModelica/pull/8791. I can see that your OpenModelica is v1.19.0. You should try the 1.19.3 release.

**Reply by Arinomo** [reply]
@mahge thanks for the info. But if i understand correctly, #8791 fixed the `[Invalid argument]' problem. Is the error code i mentioned above really caused by this bug? When i omitted the absolute path, the simulation was running until ca. 19% progress and then crashed.

**Reply by mahge** [reply]
> @mahge thanks for the info. But if i understand correctly, #8791 fixed the `[Invalid argument]' problem. Is the error code i mentioned above really caused by this bug? 

The fix was just for the input path issue. I cannot say if the crash is related to that or not. I think you should provide something we can reproduce. A minimum working example that shows the issue.  Is the library you are working with publicly available?

**Reply by Arinomo** [reply]
@mahge no, it was our own library/mode, I'll try to make a mwe for this. 

A tangent question in case i can't make a generic mwe, how can one upload a mwe involving confidential code or IP-protected code?

**Reply by mahge** [reply]
One option is to send it privately by email in a password protected zip file. However, you can first try to make a MWE since it has more uses than just secrecy. I am thinking this might be difficult to do for this specific issue. In which case we can use the email alternative. 


In the meantime, if it is okay, can you zip just the generated simulation executable (`D:/OpenModelica/OMEdit/KS_TA_entwicklung.ContinuedSim/ContinuedSim_second_run_w_init_file.exe`) and upload it here?

**Comment by casella** [comment]
@Arinomo as a rule I would always try the nightly build first 😅

**Reply by Arinomo** [reply]
Yeah that would be the most ideal way. I know i reported issues on an old version of OMEdit/OpenModelica, but sadly i don't have admin right on my workstation and can't easily install/un-install program as often as i want it. The internal process for update takes quite some time 👯 , so I think i'll wait until 1.20 became a stable release before requesting for an update on all workstations.

**Reply by casella** [reply]
OK, no problem. 1.20.0-dev.beta1 should come out anytime soon.
